src/internal/s3.ts

```diff
@@ -37,6 +37,62 @@ export class S3Client extends AWSClient {
     });
   }
 
+  /**
+   * Creates a new S3 bucket with the specified name.
+   *
+   * To use this operation, you must have the s3:CreateBucket permission.
+   *
+   * @param {string} bucketName - The name of the bucket to create
+   * @param {CreateBucketParams?} params - Optional parameters for bucket creation
+   * @throws {S3ServiceError}
+   * @throws {InvalidSignatureError}
+   */
+  async createBucket(
+    bucketName: string,
+    params?: CreateBucketParams,
+  ): Promise<void> {
+    const method = "PUT";
+
+    let body = "";
+    if (params?.region && params.region !== "us-east-1") {
```

Comment:
Do we need to take into account global vs region specific endpoints? ([docs](https://docs.aws.amazon.com/cli/latest/reference/s3api/create-bucket.html#:~:text=General%20purpose%20buckets%20%2D%20If,S3%20User%20Guide%20.)).